pages/signup/index.tsx

```diff
@@ -27,12 +27,42 @@ const ADD_USER = gql`
 `;
 
 const SignUpPage = ({ t }: PageProps): React.ReactElement => {
-  const [formValues, setFormValues] = useState<Partial<UserInput>>({
+  interface FormValues {
+    name: {
+      firstName?: string;
+      lastName: string;
+    };
+    company?: string;
+    gender: Gender;
+    email: string;
+    password: string;
+    type: 'TALENT' | 'EMPLOYER' | 'AGENCY';
+  }
+  interface AddUserInput {
+    name: {
+      firstName?: string;
+      lastName: string;
+    };
+    company?: string;
+    gender: string;
+    email: string;
+    password: string;
+    type: 'TALENT' | 'EMPLOYER' | 'AGENCY';
```

Comment:
do we have two interfaces only because gender has a different type?
Doesn't the `addUser` resolver expect `gender` to be of type `gender`? Why do we have to transform it to be a string?

Reply:
> why do you add a second version of the package-lock file?

I may have deleted it once and then it got regenerated. Shouldn't package-lock be in our gitignore anyway?

> do we have two interfaces only because gender has a different type?
> Doesn't the `addUser` resolver expect `gender` to be of type `gender`? Why do we have to transform it to be a string?

The graphql enums are actual strings I think, whereas the typescript enums are ints that have a string value. UserInput expects to get a graphql enum, but sending the ts enum produces a graphql error, whereas sending the associated string Gender[Gender.Other] does not.

Having transformers between types used in the frontend and types used in graphql is therefore a reasonable thing, since it allows us to work with the expected types on both ends and does not require case by case conversion. In the long run, these transformers should go into their own file, though, I would say.

Reply:
> I may have deleted it once and then it got regenerated. Shouldn't package-lock be in our gitignore anyway?
> 
I wouldn't put it to gitignore by default. It might make sense to make changes to the package-lock if you are e.g. adding new packages to package.json. Then, if the package-lock is updated, one does not have to install all these packages each time.

> 
> The graphql enums are actual strings I think, whereas the typescript enums are ints that have a string value. UserInput expects to get a graphql enum, but sending the ts enum produces a graphql error, whereas sending the associated string Gender[Gender.Other] does not.
> 
> Having transformers between types used in the frontend and types used in graphql is therefore a reasonable thing, since it allows us to work with the expected types on both ends and does not require case by case conversion. In the long run, these transformers should go into their own file, though, I would say.

Okay, I understand. Fine for me.